main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  # my-each
+
+  def my_each
+    ind = 0
+
+    while ind != length
+      yield self[ind]
+      ind += 1
+    end
+
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    index = 0
+    my_each do |el|
+      yield(el, index)
+      index += 1
+    end
+
+    self
+  end
+
+  # my_select
+
+  def my_select
+    filtered_arr = []
+
+    while length != 0
+      yield(first) && filtered_arr.push(first)
+      shift
+    end
+
+    filtered_arr
+  end
+
+  # my_all
+
+  def my_all?
+    output = true
+
+    while length != 0
+      if !yield(first)
+        output = false
+        break
+      else
+        shift
+      end
+    end
+    output
+  end
+
+  # my_any
+
+  def my_any?
+    output = false
+    my_each do |el|
+      if yield el
+        output = true
+        break
+      end
+    end
+    output
+  end
+
+  # my_none
+
+  def my_none?
+    output = true
+
+    while length != 0
+      if yield first
+        output = false
+        break
+      else
+        shift
+      end
+    end
+    output
+  end
+
+  # my_count
+
+  def my_count
+    count = 0
+
+    my_each do |el|
+      yield(el) && count += 1
+    end
+    count
+  end
+
+  # my_map
+
+  def my_map(&proc_block)
+    new_arr = []
+
+    my_each { |el| new_arr.push(proc_block.call(el)) }
+
+    new_arr
+  end
```

Comment:
- [ ] should have working `my_map` **(covered by automated tests)** identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should return an enumerator if no block or argument is given.
- It should work with ranges.
- It should execute only the proc when both a block and a proc are given (for each element).